pytorch_lightning/utilities/types.py

```diff
@@ -46,12 +46,11 @@
 EVAL_DATALOADERS = Union[DataLoader, Sequence[DataLoader]]
 
 
-# Copied from `torch.optim.lr_scheduler.pyi`
-# Missing attributes were added to improve typing
+# Inferred from `torch.optim.lr_scheduler.pyi`
 class _LRScheduler:
     optimizer: Optimizer
 
-    def __init__(self, optimizer: Optimizer, last_epoch: int = ...) -> None:
+    def __init__(self, optimizer: Optimizer, *args: Any, **kwargs: Any) -> None:
```

Comment:
why is it necessary to deviate from the pytorch definitions?

Reply:
the main motivation is to support custom schedulers that follow certain protocol API, but not necessarily the arguments. One example: https://github.com/rwightman/pytorch-image-models/blob/master/timm/scheduler/scheduler.py